### ISSUE TITLE
PaLM apiEndpoint 404 error

### ISSUE THREAD

**Issue by eostis** [weaviate/weaviate-io#784]
Hi,

The documentation https://weaviate.io/developers/weaviate/modules/retriever-vectorizer-modules/text2vec-palm#schema-configuration mentions an apiEndpoint:
```
{
  "classes": [
    {
      "class": "Document",
      "description": "A class called document",
      "vectorizer": "text2vec-palm",
      "moduleConfig": {
        "text2vec-palm": {
          "apiEndpoint": "YOUR-API-ENDPOINT",             // Required. Replace with your value.
          "projectID": "YOUR-GOOGLE-CLOUD-PROJECT-ID",    // Required. Replace with your value.
          "modelId": "YOUR-GOOGLE-CLOUD-MODEL-ID",        // Optional. Defaults to `textembedding-gecko-001`.
        },
      },
    }
  ]
}

```

### What would be the apiEndpoint above, considering the 2 documentations below ?

- In a Vertex AI Google Cloud project:
```
ENDPOINT_ID="textembedding-gecko-001"
PROJECT_ID="cloud-large-language-models"

curl -X POST \
  -H "Authorization: Bearer $(gcloud auth print-access-token)" \
  -H "Content-Type: application/json" https://us-central1-aiplatform.googleapis.com/v1/projects/${PROJECT_ID}/locations/us-central1/endpoints/${ENDPOINT_ID}:predict -d \
  '{"instances": [{"content": "listening to classical music"}]}'

```

- In Generative AI documentation https://developers.generativeai.google/api/rest/generativelanguage/models/embedText:
```
curl https://generativelanguage.googleapis.com/v1beta2/models/embedding-gecko-001:embedText?key=$PALM_API_KEY \
  -H 'Content-Type: application/json' \
  -X POST \
  -d '{"text": "say something nice!"}'
```

**Comment by databyjp** [comment]
Thanks @etiennedi I've moved it.

@eostis - I've updated the parameters (deploying now). And the example that I received had these parameters. Does that help? If not I can get someone with PaLM access to give me more of a pointer.

I'll add some example like this to the docs also

```
  "moduleConfig": {
    "generative-palm": {
      "apiEndpoint": "us-central1-aiplatform.googleapis.com",
      "projectId": "cloud-large-language-models",
      "endpointId": "4511608470067216384"
    },
    "text2vec-palm": {
      "apiEndpoint": "us-central1-aiplatform.googleapis.com",
      "projectId": "cloud-large-language-models",
      "modelId": "textembedding-gecko-001"
    }
  },
```

**Comment by eostis** [comment]
I use the same endpoint, but get the same error.

I could reproduce the error with:
```
TOKEN="(token generated on Google shell console)"
ENDPOINT_ID="textembedding-gecko-001"
PROJECT_ID="weaviate-374814"

curl -X POST   -H "Authorization: Bearer $TOKEN"   -H "Content-Type: application/json" https://us-central1-aiplatform.googleapis.com/v1/projects/${PROJECT_ID}/locations/us-central1/endpoints/${ENDPOINT_ID}:predict -d   '{"instances": [{"content": "listening to classical music"}]}' -v

* Hostname was NOT found in DNS cache
*   Trying 172.217.20.202...
* Connected to us-central1-aiplatform.googleapis.com (172.217.20.202) port 443 (#0)
* SSLv3, TLS handshake, Client hello (1):
* SSLv3, TLS handshake, Server hello (2):
* SSLv3, TLS handshake, CERT (11):
* SSLv3, TLS handshake, Server finished (14):
* SSLv3, TLS handshake, Client key exchange (16):
* SSLv3, TLS change cipher, Client hello (1):
* SSLv3, TLS handshake, Finished (20):
* SSLv3, TLS change cipher, Client hello (1):
* SSLv3, TLS handshake, Finished (20):
* SSL connection using TLSv1.0 / AES128-SHA
* Server certificate:
*        subject: CN=upload.video.google.com
*        start date: 2023-04-24 12:00:43 GMT
*        expire date: 2023-07-17 12:00:42 GMT
*        subjectAltName: us-central1-aiplatform.googleapis.com matched
*        issuer: C=US; O=Google Trust Services LLC; CN=GTS CA 1C3
*        SSL certificate verify ok.
> POST /v1/projects/weaviate-374814/locations/us-central1/endpoints/textembedding-gecko-001:predict HTTP/1.1
> User-Agent: curl/7.38.0
> Host: us-central1-aiplatform.googleapis.com
> Accept: */*
> Authorization: Bearer (token generated on Google shell console)
> Content-Type: application/json
> Content-Length: 60
> 
* upload completely sent off: 60 out of 60 bytes
< HTTP/1.1 404 Not Found
< Vary: X-Origin
< Vary: Referer
< Content-Type: application/json; charset=UTF-8
< Date: Sun, 14 May 2023 13:13:57 GMT
* Server scaffolding on HTTPServer2 is not blacklisted
< Server: scaffolding on HTTPServer2
< Cache-Control: private
< X-XSS-Protection: 0
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
< Accept-Ranges: none
< Vary: Origin,Accept-Encoding
< Transfer-Encoding: chunked
< 
{
  "error": {
    "code": 404,
    "message": "Endpoint `projects/weaviate-374814/locations/us-central1/endpoints/textembedding-gecko-001` not found.",
    "status": "NOT_FOUND"
  }
}
* Connection #0 to host us-central1-aiplatform.googleapis.com left intact

```

**Comment by eostis** [comment]
No error with the other endpoint:
```
curl -X POST \
  -H "Authorization: Bearer $TOKEN" \
  -H "Content-Type: application/json" https://us-central1-aiplatform.googleapis.com/v1/projects/${PROJECT_ID}/locations/us-central1/endpoints/publishers/google/models/${ENDPOINT_ID}:predict -d \
  '{"instances": [{"content": "listening to classical music"}]}'


```

**Comment by databyjp** [comment]
Hmm. I'm afraid I don't quite know what's going on there. 

@antas-marcin when you have time could you take a look please? 🙏

**Comment by eostis** [comment]
Same example from the Google shell console:
<img width="1437" alt="image" src="https://github.com/weaviate/weaviate-io/assets/1785759/b63fb7fb-aa54-49b0-b36b-e3f37a97f781">

**Comment by eostis** [comment]
This url worked out (from https://cloud.google.com/vertex-ai/docs/generative-ai/embeddings/get-text-embeddings#curl):
```
curl -X POST \
  -H "Authorization: Bearer $(gcloud auth print-access-token)" \
  -H "Content-Type: application/json" https://us-central1-aiplatform.googleapis.com/v1/projects/${PROJECT_ID}/locations/us-central1/publishers/google/models/${ENDPOINT_ID}:predict -d \
  '{"instances": [{"content": "listening to classical music"}]}'
```

<img width="806" alt="image" src="https://github.com/weaviate/weaviate-io/assets/1785759/2eb438a5-16ca-4648-8a87-eab34afd0695">

**Comment by databyjp** [comment]
Okay, excellent. Did you get the module to work? 

Do the parameters as they are documented make sense?

**Comment by antas-marcin** [comment]
@eostis you're right, we are building different URL's, I'll make a fix tomorrow for this. Maybe even I'll be able to create a patch release for that. I will keep you posted

**Comment by antas-marcin** [comment]
@eostis I have a [PR](https://github.com/weaviate/weaviate/pull/3033) ready to be merged with a fix for your issue

**Comment by antas-marcin** [comment]
@eostis we have released [v1.19.4](https://github.com/weaviate/weaviate/releases/tag/v1.19.4) which should resolve your issue.

Also now configuration of PaLM modules is simplified, now only `projectId` setting is mandatory, rest of the settings are optional look [here](https://weaviate.io/developers/weaviate/modules/retriever-vectorizer-modules/text2vec-palm) and [here](https://weaviate.io/developers/weaviate/modules/reader-generator-modules/generative-palm) for details.
